website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/Always.yaml`
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/IfNotPresent.yaml`
+- `testdata/ansible/memcached-operator/config/testing/pull_policy/Never.yaml`
+
+ Ansible operators may also create Pod-defining objects dynamically. An example of this kind of object would be in `testdata/ansible/memcached-operator/roles/memcached/tasks/main.yml`.
+ ```yaml
+# tasks file for Memcached
+- name: start memcached
+  kubernetes.core.k8s:
+    definition:
+      ...
+      spec:
+        ...
+        template:
+          ...
+          spec:
+            containers:
+            - name: memcached
+              securityContext:
+                allowPrivilegeEscalation: false
+                capabilities:
+                  drop:
+                    - "ALL"
+              command:
+              - memcached
+              - -m=64
+              - -o
+              - modern
+              - -v
+              image: "docker.io/memcached:1.4.36-alpine"
+              ports:
+                - containerPort: 11211
+              readinessProbe:
+                tcpSocket:
+                  port: 11211
+                initialDelaySeconds: 3
+                periodSeconds: 3
+ ```
+
+ Because the is an embedded yaml, this can be updated by inserting an `affinity` block directly into the yaml like so.
+ ```yaml
+# tasks file for Memcached
+- name: start memcached
+  kubernetes.core.k8s:
+    definition:
+      ...
+      spec:
+        ...
+        template:
+          ...
+          spec:
+            affinity:
+              nodeAffinity:
+                requiredDuringSchedulingIgnoredDuringExecution:
+                  nodeSelectorTerms:
+                  - matchExpressions:
+                    - key: kubernetes.io/arch
+                      operator: In
+                      values:
+                      - amd64
+                      - arm64
+                      - ppc64le
+                      - s390x
+            containers:
+              ...
+ ```
+
+#### Go Operators
+In the `testdata/go/v3` samples app, you'll find several configuration objects that need a nodeAffinity specification in the `config` and `bundle/manifests` directories. The full list is as follows:
+- `testdata/go/v3/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_config_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_webhook_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/default/manager_auth_proxy_patch.yaml`
+- `testdata/go/v3/memcached-operator/config/manager/manager.yaml`
+
+ Go operators may also create Pod-defining objects dynamically. An example of this kind of object would be in `testdata/go/v3/memcached-operator/controllers/memcached_controller.go`.
+```go
+Spec: corev1.PodSpec{
+    SecurityContext: &corev1.PodSecurityContext{
+      ...
```

Comment:
Why do we need this part? Why not just add "Following an example using Golang for you know how can you configure the nodeAffinity for the workloads in the reconciliation"

Reply:
So my thinking was:
1. Provide the general idea and the common ground between the operator types
2. Provide a before and after walkthrough to explain how those changes might look different for each of the operator types when those objects are defined in the different languages.

I could absolutely simplify the whole back-half of the doc down to - we leave the specifics of the transformation to include node-affinity in your `Pod` definitions as an exercise for the user.

However, I guess I was writing the document more for people like me. I don't know the specific syntactical differences between languages (and would struggle to even know where to look for such info), so I find it helpful to see an example where I can see exactly what changed so that I can replicate it each place where I can identify the pattern.